packages/sdk-react/src/components/receivables/InvoiceDetails/CreateReceivable/CreateReceivables.test.tsx

```diff
@@ -44,37 +57,100 @@ describe('CreateReceivables', () => {
-    triggerClickOnSelectOption(
-      /Bill to/i,
-      getCounterpartName(firstCounterpart)
+    await waitForCondition(
+      () =>
+        !!screen.queryByRole('button', {
+          name: t`Next page`,
+        })
     );
+    const nextPageButtonPromise = screen.findByRole('button', {
+      name: t`Next page`,
+    });
+    await expect(nextPageButtonPromise).resolves.toBeEnabled();
 
-    // Select the first two items
-    fireEvent.click(screen.getByRole('button', { name: 'Add item' }));
-    const checkboxes = await screen.findAllByRole('checkbox');
-    fireEvent.click(checkboxes[1]);
-    fireEvent.click(checkboxes[2]);
-    fireEvent.click(screen.getByRole('button', { name: 'Add' }));
-
-    await waitFor(() => {
-      const submitButton = screen.getByRole('button', { name: /create/i });
-      fireEvent.click(submitButton);
+    // No contact progress indicator should be visible at this moment
+    const contactPersonPromise = screen.findByLabelText<HTMLInputElement>(
+      t`Contact person`
+    );
+    await expect(contactPersonPromise).resolves.toBeInTheDocument();
+    const contactPersonInput = await contactPersonPromise;
+    expect(
+      contactPersonInput.parentElement!.querySelector(
+        '.MuiCircularProgress-root'
+      )
+    ).toBeNull();
+
+    await triggerClickOnFirstAutocompleteOption(/Bill to/i);
+    await triggerClickOnFirstAutocompleteOption(/Billing address/i);
+    await triggerClickOnFirstAutocompleteOption(/Your VAT ID/i);
+    await triggerClickOnFirstAutocompleteOption(/Payment terms/i);
+
+    // Add item to invoice
+    const itemsHeader = screen.getByText(t`Items`);
+    // Use querySelector to find the 'Add Item' button.
+    // Due to the button icon, the screen.getByText won't find it
+    const addItemButton = itemsHeader.parentElement!.querySelector('button')!;
+    fireEvent.click(addItemButton);
+    // Wait for Products dialog to open
+    await waitForCondition(
+      () => !!screen.queryByText(`Available items`),
+      3_000
+    );
 
-      expect(onCreateMock).toHaveBeenCalled();
+    const availableItems = screen.getByText(t`Available items`);
+    const rightSideForm = findParentElement(
+      availableItems,
+      ({ tagName }) => tagName == 'FORM'
+    )!;
+    // Wait for products to load
+    await waitForCondition(
+      () => !!rightSideForm.querySelector('tbody tr input'),
+      3_000
+    );
+    const productCheckbox = rightSideForm.querySelector('tbody tr input')!;
+    fireEvent.click(productCheckbox);
+
+    const addProductsButton = screen.getByRole('button', { name: t`Add` });
+    fireEvent.click(addProductsButton);
+    // Wait for Products dialog to close
+    await waitForElementToBeRemoved(addProductsButton, {
+      timeout: 3_000,
     });
-  });
+
+    // Create invoice
+    const nextPageButton = await nextPageButtonPromise;
+    fireEvent.click(nextPageButton);
+
+    // Compose email dialog opens
+    await waitForCondition(
```

Comment:
Can we please make some cleanups here and remove // comments as code does looks as self explanatory

Reply:
I removed comments as suggested